src/transformers/models/flava/processing_flava.py

```diff
@@ -0,0 +1,146 @@
+# coding=utf-8
+# Copyright 2021 Meta Platforms authors and The HuggingFace Team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Image/Text processor class for FLAVA
+"""
+from typing import List, Optional, Union
+
+import numpy as np
+
+from transformers.data.data_collator import DataCollatorForWholeWordMask, tolist
+
+from ...processing_utils import ProcessorMixin
+from ...tokenization_utils_base import BatchEncoding, PaddingStrategy, PreTokenizedInput, TextInput, TruncationStrategy
+from ...utils import TensorType
+
+
+class FLAVAProcessor(ProcessorMixin):
+    r"""
+    Constructs a FLAVA processor which wraps a FLAVA feature extractor and a FLAVA tokenizer into a single processor.
+
+    [`FLAVAProcessor`] offers all the functionalities of [`FLAVAFeatureExtractor`] and [`FLAVATokenizerFast`]. See the
+    [`~FLAVAProcessor.__call__`] and [`~FLAVAProcessor.decode`] for more information.
+
+    Args:
+        feature_extractor ([`FLAVAFeatureExtractor`]):
+            The feature extractor is a required input.
+        tokenizer ([`FLAVATokenizerFast`]):
+            The tokenizer is a required input.
+    """
+    feature_extractor_class = "FLAVAFeatureExtractor"
+    tokenizer_class = ("BertTokenizer", "BertTokenizerFast")
+
+    def __init__(self, feature_extractor, tokenizer, mlm_probability=0.15):
+        super().__init__(feature_extractor, tokenizer)
+        self.current_processor = self.feature_extractor
+        self.text_masker = DataCollatorForWholeWordMask(tokenizer, mlm=True, mlm_probability=mlm_probability)
```

Comment:
This class shouldn't be used at all, even less here (it's community contributed and only works for BERT). Data collation is outside of the scope of processors. It also doesn't seem to be used inside this processor, so should be defined by the user independently when they need it.

Reply:
Big +1 from my side here. I don't think we should add training code to a processor clas. The `input_ids_masked` should be passed directly to the model in the correct form and not be created here

Reply:
I am only using it to mask the text not actually collating the data. I can write the logic on my own but would prefer to reuse the existing code. I feel like it is very cumbersome as of now to create *masked* `input_ids` in `transformers` except doing it through DataCollators. Even in data collators, they won't return the original `input_ids` which we need for FLAVA. I would ideally like my processor to return masked input ids if `return_mask` is passed so that i can create an e2e script for testing masking. Happy to hear suggestions on how it could be done without Data Collator. Alternatively, do you suggest creating a new datacollator for FLAVA?